### ISSUE TITLE
Unknown channel option 'io.netty.channel.unix.UnixChannelOption#SO_REUSEPORT'

### ISSUE THREAD

**Issue by azagniotov** [eclipse/vert.x#2193]
I am running a service using Vert.x `v3.5.0` with `setPreferNativeTransport(true)` in `VertxOptions`. I am seeing periodic warnings in the logs:
```
2017-10-30 xx:xx:xx,xxx WARN  [vert.x-acceptor-thread-0] ServerBootstrap: Unknown channel option 'io.netty.channel.unix.UnixChannelOption#SO_REUSEPORT' for channel '[id: 0xd5573f68, L:/10.xxx.xxx.xxx:8080 - R:/10.xxx.xxx.xxx:25442]'
```

Please advise

**Comment by azagniotov** [comment]
I did a little more verification and saw that this warning definitely comes up when `setPreferNativeTransport(true)`. 

I am not sure how much of a problem this warning is when native support is enabled and whether it is a problem per-se.

**Comment by vietj** [comment]
it's not really a problem per se (only a logging cosmetic issue), can you tell what is your OS and what options you set ?

**Comment by azagniotov** [comment]
Hi Julien,

Options for Vert.x? The `setPreferNativeTransport(true)` for the only one set, everything else is default

`OS` => 
Linux version 3.10.0-514.16.1.el7.x86_64 (builder@kbuilder.dev.centos.org) (gcc version 4.8.5 20150623 (Red Hat 4.8.5-11) (GCC) ) #1 SMP Wed Apr 12 15:04:24 UTC 2017

`Kernel` => 
Linux xxx 3.10.0-514.16.1.el7.x86_64 #1 SMP Wed Apr 12 15:04:24 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

**Comment by vietj** [comment]
we will resolve this, in the meantime just ignore it, it's a warning

normally your Docker container should work well with native although I haven't checked formally.

**Comment by azagniotov** [comment]
Thanks!

**Comment by zyclonite** [comment]
i see this warning as well... docker (image: openjdk 1.8.0_144) as well as on linux 4.13.x kernel (same java version)

does not have any impact on the native implementation as it seems

**Comment by vietj** [comment]
no, it's more a socket option that cannot be set on this implementation and there is therefore a warn, vertx should avoid to set this option for this implementation

**Comment by shenzhenMirren** [comment]
@vietj

**Comment by shenzhenMirren** [comment]
@vietj 
I have the same problem,
my os 
[root@duhua home]# uname -a
Linux duhua 3.10.0-693.2.2.el7.x86_64 #1 SMP Tue Sep 12 22:26:13 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux
[root@duhua home]# lsb_release -a
LSB Version:	:core-4.1-amd64:core-4.1-noarch
Distributor ID:	CentOS
Description:	CentOS Linux release 7.4.1708 (Core) 
Release:	7.4.1708
Codename:	Core
![qq 20180124155007](https://user-images.githubusercontent.com/24805209/35320303-42f60914-011e-11e8-90ca-84c4d0ac86cf.png)

**Comment by guoyu511** [comment]
@vietj 
I have see the same warning on my GCE machines, CentOS 7.

Actually it caused by a incorrect usage of SO_REUSEPORT at here:

https://github.com/eclipse/vert.x/blob/c79c698d2dc36f26245e4c390c0db755eb1a91fd/src/main/java/io/vertx/core/net/impl/transport/Transport.java#L244

The SO_REUSEPORT option should be set to the ServerBootstrap it self, not the child channels.

I have made a test to reproduce this problem:
https://github.com/guoyu511/vertx-native-transport-test

Here is a PR for this issue, hope any one can review it:
[https://github.com/eclipse/vert.x/pull/2358](url)

**Comment by michaelpog** [comment]
I see the same warning all over my logs when I set ``` setPreferNativeTransport(true) ```
No matter if I set the reusePort to true or false on the HTTP server options.
My OS is CentOS 6.7

**Comment by vietj** [comment]
have you tested with current 3.5 branch @michaelpog that should be fixed

**Comment by manikantag** [comment]
I can confirm with Vertx 3.5.1 & Netty 4.1.25-Final, I'm getting this error.

```
...
2018-05-23 07:08:48.275 +0000 [vert.x-eventloop-thread-3] WARN  io.netty.bootstrap.Bootstrap - Unknown channel option 'io.netty.channel.unix.UnixChannelOption#SO_REUSEPORT' for channel '[id: 0x3621f9da]'
2018-05-23 07:08:48.276 +0000 [vert.x-eventloop-thread-3] WARN  io.netty.bootstrap.Bootstrap - Unknown channel option 'io.netty.channel.unix.UnixChannelOption#SO_REUSEPORT' for channel '[id: 0xec482c3c]'
2018-05-23 07:08:48.276 +0000 [vert.x-eventloop-thread-3] WARN  io.netty.bootstrap.Bootstrap - Unknown channel option 'io.netty.channel.unix.UnixChannelOption#SO_REUSEPORT' for channel '[id: 0xe69ab6ab]'
...
```

**Comment by vietj** [comment]
@manikantag it should be gone with 3.5.2.CR2

**Comment by manikantag** [comment]
@vietj Oh, I'll check. When can we expect 3.5.2 GA?

**Comment by vietj** [comment]
by end of this month top

**Comment by manikantag** [comment]
@vietj Oh, nice.
